### DISCUSSION TITLE
METviewer Taylor Diagram graphical values do not match the R data values

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#2549] [Plot Generation] [answered]
I will provide the plots and plot_spec files to Julie Prestopnik by email for further distribution within DTC as needed. The R data I have is as follows:
model	          valid_hour	stat_name	fcst_var	stat_value	            stat_btcl    stat_btcu	nstats
URMA2pt5_d01	0	FBAR	TMP	288.899386241959	NA	NA	3
URMA2pt5_d01	0	FSTDEV	TMP	6.70492916311306	NA	NA	3
URMA2pt5_d01	0	OBAR	TMP	288.918529804854	NA	NA	3
URMA2pt5_d01	0	OSTDEV	TMP	6.74934581722694	NA	NA	3
URMA2pt5_d01	0	PR_CORR	TMP	0.986896213222442	NA	NA	3
URMA2pt5_d01	0	RMSE	TMP	1.09010235655628	NA	NA	3

Maybe I'm reading the diagram incorrectly, but for example the FSTDEV value of 6.7 doesn't match the scale values of the standard deviation along the x and y axes. 6.7 would be plotted well outside of the diagram as scaled. Also the RMSE value of 1.09 doesn't make sense when the red dot data point is well inside the radius of 0.5 from the "observation" location on the x-axis. The value of the Pearson Correlation Coef from the diagram looks reasonable compared to the value from the R data. Can you explain what I'm doing wrong interpreting the data from the diagram or diagnose the error I'm making when plotting the diagram? Thanks.

**Comment by bikegeek** [comment]
I will need to take a look at your input data and see if I can reproduce your results and compare those to the output points from the Python implementation in METplotpy.  METviewer is now using the Python implementation of the Taylor diagram.

**Reply by jwraby** [reply]
Thanks for your quick response! I just sent the plot_spec and PNG files by email to Julie. I can't upload the files into this discussion. Do you need any other data such as the STAT files.

**Reply by jprestop** [reply]
@bikegeek I have put the following files at seneca:/d1/projects/METplus/discussions/2549:
```
plot_20240410_121652.png  
plot_20240410_121652.xml  
point_stat_URMA_MSA_REGION_d01_000000L_20161112_000000V.stat
point_stat_URMA_MSA_REGION_d01_000000L_20161113_000000V.stat
point_stat_URMA_MSA_REGION_d01_000000L_20161114_000000V.stat
```

**Comment by bikegeek** [comment]
Thanks for providing the data and other results.  I have a question:

The output from METviewer indicates that you have the correct statistics needed to plot the Taylor Diagram (FSTDEV, OSTDEV, and PR_CORR), but the .stat files you provided (via Julie) are entirely MPR linetype.  

Are you doing some processing to the MPR output before loading it into the METviewer database?
How are you obtaining the FSTDEV, OSTDEV, and PR_CORR statistics from your data?

The FSTDEV, OSTDEV, and PR_CORR statistics are in the CNT linetype, which is what I was anticipating for input data.

Regards,
Minna

**Comment by jwraby** [comment]
Minna -
Thanks for your response.

I opened up one of the STAT files resident on the local linux workstation which is where METviewer runs and when I search for strings "CNT",  "SL1L2", and "MPR" I find all 3 present. I checked the Point-Stat config file I used to produce them and I specified the output of all 3 line types using "BOTH" for each one.

I'm not doing any processing of the STAT files prior to loading them into the database. After the diagram is plotted, checking the R data tab shows that all the needed statistics you list are present. See the dialog and data I input when I first created the discussion. I can try uploading and resending the 3 STAT files, but I won't be able to do this now as my system is having network issues. I'm not working tomorrow but will be back on Mon morning.

R/
John
________________________________

**Comment by bikegeek** [comment]
Thanks John, I don't think you need to re-run anything.  Apologies, I
searched your data and found the CNT.  I'm multi-tasking with another
project and was looking for ECNT instead of CNT in your input.  I'll
attempt to generate the Taylor diagram via command line (METplotpy) and see
what data and plot it produces.

Regards,
Minna

**Comment by jwraby** [comment]
Understand completely, Minna. No apology needed.
R/
John
________________________________

**Comment by bikegeek** [comment]
I am reproducing a plot that is similar to the one generated by METviewer, with similar values that you produced.  I'll need to check the formula used by the plot to see why the points don't appear where they should.

**Reply by jwraby** [reply]
Thanks for the update.

**Comment by bikegeek** [comment]
I looked at the equations, and the stdev is being normalized, which is why we aren't observing the 6.x values we see in the R data output:

stdev = FSTDEV/OSTDEV 

stdev = 6.704/6.749
          = 0.993

and the Pearson's correlation is plotted without modification, which explains why the correlation on the plot is consistent with the R data value

**Reply by jwraby** [reply]
This is good information. I wasn't aware of the normalization of the stdev. I looked in the Rscript tab to see if the script has that computation, but I didn't see it. Must be there because the location of the plotted red data point corresponds to the value of 0.993 on the Standard Deviation scale of the plot. Were you able to find out why the RMSE value doesn't appear to be plotted correctly? Looking at the plot it looks like the value should be about 0.2, but the R data value is 1.09.

**Comment by bikegeek** [comment, marked as answer]
The R script tab no longer has any useful information, the R implementation is no longer used. The Python implementation of the Taylor Diagram is the only available means of plotting.  

The RMS value isn't directly plotted. The pr_corr and stdev (normalized) are the only two values used to plot the point and the grid lines for the RMS represent the centered RMS difference.  The RMS difference is related to the correlation and stdev by this relationship (from Taylor's paper):

      

     E^2 = sigmaf^2 _ sigmar^2 - (2sigmaf*sigmar *R)

        sigmaf^2 and sigmar^2 are variaces of the fcst and reference fields
        R is correlation coeff
        E is the centered RMS difference

So in a nutshell, the RMS value on the Taylor diagram is **supposed** to reflect this relationship.

**Reply by jwraby** [reply]
This is very informative. I really need to read the Taylor paper to better understand the diagram. I think the description in the METviewer User's Guide, which actually cites the paper, was very brief and referred to the basic statistics "RMSE", Standard Deviation" and "Pearson Correlation Statistic" which I interpreted differently from the way that Taylor was using the statistics to produce the diagram. See the User's Guide: https://metviewer.readthedocs.io/en/main_v4.1/Users_Guide/taylordiag.html. In retrospect, I should have read the paper to get a better understanding of the diagram before asking all the questions. I really appreciate all your investigation into the diagram which did answer all my questions.

**Comment by bikegeek** [comment]
Thanks for the feedback, I should get a statistician to add more content to
the User's Guide/fill in more background information.  I'll create a Github
issue for documentation, but it won't make it into the next release:

     https://github.com/dtcenter/METplotpy/issues/435

Please feel free to make any additional comments on what you would find
useful in the documentation under the 'comments' section of the issue.

Regards,
Minna
